core/preconditioner/jacobi.cpp

```diff
@@ -343,12 +343,12 @@ void Jacobi<ValueType, IndexType>::generate(const LinOp* system_matrix,
             diag = share(as<DiagonalLinOpExtractable>(system_matrix)
                              ->extract_diagonal_linop());
         }
-        auto diag_vt =
-            ::gko::detail::temporary_conversion<matrix::Diagonal<ValueType>>::
-                template create<matrix::Diagonal<previous_precision<ValueType>>,
-                                matrix::Diagonal<previous_precision<
-                                    previous_precision<ValueType>>>>(
-                    diag.get());
+        auto diag_vt = ::gko::detail::
+            temporary_conversion<matrix::Diagonal<ValueType>>::template create<
+                matrix::Diagonal<previous_precision<ValueType>>,
+                matrix::Diagonal<previous_precision_move<ValueType, 2>>,
+                matrix::Diagonal<previous_precision_move<ValueType, 3>>>(
```

Comment:
I think this needs to be guarded.

Reply:
It only gives some redundant check with the fixed `*_move` template.
The code using the redundant part means nothing matched here, so it will throw the error in the end.